docs/_docs/cross-chain/core-messaging-layer.md

```diff
@@ -0,0 +1,59 @@
+---
+---
+# Obscuro Cross Chain Messaging
+
+Obscuro is equipped with a cross chain messaging protocol that enables sending and receiving messages across layers securely and in an entirely decentralized fashion.
+
+The core contract that provides this functionality is the `MessageBus`. It exists both on L1 and L2. In the L1 you can find it under the management contract whilst on the L2 it is created and managed by the enclave.
+
+## How It Works
+
+Users call the `publishMessage` function on the `MessageBus`. This function emits an event that records what message is being published, who is calling the method and some other bits. This event later on gets picked on by the enclave which ensures that the message is available on the counterpart `MessageBus`.
+
+When the message is made available, users can call the `verifyMessageFinalized` method in the `MessageBus` on the receipient layer. This method will return `true` if the message was indeed sent on the other layer as presented. Same `msg.sender`, same `payload`, same `nonce`.
```

Comment:
Is it worth saying you effectively need to construct the cross chain message from the transaction logs in order to call verifyMessageFinalized? Basically are we trying to keep this doc fairly high level or give more granular detail of the various call structures.

Reply:
Yep, trying for this to be fairly high level. My expectation is that an automated relayer will pop up at some point so the low level detail will be hidden away.